src/nsrt_learning/strips_learning/gen_to_spec_learner.py

```diff
@@ -78,6 +78,9 @@ def _learn(self) -> List[PartialNSRTAndDatastore]:
 
         # Finish learning by adding in the delete effects and side predicates.
         all_pnads = self._finish_learning(param_opt_to_nec_pnads)
+
+        # Recompute datastores one final time.
+        self._recompute_datastores_from_segments(all_pnads)
```

Comment:
i don't think this is necessary but i got scared since now i'm messing with add effects and preconds in finish_learning()

Reply:
update: this is necessary with some changes that are coming up